src/v/features/feature_table.cc

```diff
@@ -423,6 +423,20 @@ void feature_table::testing_activate_all() {
     on_update();
 }
 
+feature_table::version_fence
+feature_table::decode_version_fence(model::record_batch batch) {
+    auto records = batch.copy_records();
```

Comment:
did you mean something else, because `record_batch::release_records` doesn't exist. copy records, despite the name, doesn't really copy records that already exist, it decodes records from the internal iobuf into a vector as a convenience over using the for_each_record pattern with a lambda.